tests/core/test_stellargraph.py

```diff
@@ -738,3 +738,13 @@ def test_info_heterogeneous():
 
     assert " A-R->B: [5]"
     assert " B-F->B: [1]"
+
+
+def test_edges_include_weights():
+    g = example_weighted_hin()
+    edges, weights = g.edges(include_edge_weight=True)
+    nxg = g.to_networkx()
+    assert len(edges) == len(weights) == len(nxg.edges())
+    for (src, dst), w in zip(edges, weights):
+        # check weight matches an edge in multi-edges
+        assert w in {data["weight"] for data in nxg.get_edge_data(src, dst).values()}
```

Comment:
This set is going to be recomputed for every edge, so it's probably mildly cleaner to avoid doing that construction:

```python
assert any(data["weight"] == w for data in ...)
```

Another option would be to group by the `src, dst` pairs and compare the complete lists, because, as is, this is only verifying that it's returning a weight between the edges, not every weight between them. This could be built on pandas like `pd.DataFrame(edges, columns=["source", "target"]).assign(weight=weights).groupby(["source", "target"]).agg(list)` to get a data frame mapping the source/target pairs to list of weights, and then iterate through it or something, but that's kinda complicated.

Reply:
I think I'll just do your pandas suggestion